#!/usr/bin/env python
PACKAGE = "line_locator"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("h_low", int_t, 0, "Hue Lower", 0, 0, 255)
gen.add("h_high", int_t, 0, "Hue Upper", 255, 0, 255)

gen.add("s_low", int_t, 0, "Saturation Lower", 0, 0, 255)
gen.add("s_high", int_t, 0, "Saturation Upper", 255, 0, 255)

gen.add("v_low", int_t, 0, "Value Lower", 0, 0, 255)
gen.add("v_high", int_t, 0, "Value Upper", 255, 0, 255)

gen.add("area_low", double_t, 0, "Area Lower", 0, 0, 100000)
gen.add("area_high", double_t, 0, "Area Upper", 100000, 0, 100000)

gen.add("circularity_low", double_t, 0, "Circularity Lower", 0., 0, 1.0)
gen.add("circularity_high", double_t, 0, "Circularity Upper", 1.0, 0, 1.0)

exit(gen.generate(PACKAGE, "line_locator", "ImageAnalysis"))
